hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-shuffle/src/main/java/org/apache/hadoop/mapred/ShuffleHandler.java

```diff
@@ -785,37 +833,54 @@ private void removeJobShuffleInfo(JobID jobId) throws IOException {
     }
   }
 
-  static class TimeoutHandler extends IdleStateAwareChannelHandler {
+  @VisibleForTesting
+  public void setUseOutboundExceptionHandler(boolean useHandler) {
+    this.useOutboundExceptionHandler = useHandler;
+  }
 
+  static class TimeoutHandler extends IdleStateHandler {
+    private final int connectionKeepAliveTimeOut;
     private boolean enabledTimeout;
 
+    public TimeoutHandler(int connectionKeepAliveTimeOut) {
+      //disable reader timeout
+      //set writer timeout to configured timeout value
+      //disable all idle timeout
+      super(0, connectionKeepAliveTimeOut, 0);
```

Comment:
This was a bit confusing for me, and the netty documentation didn't help, so I think it might be a good ide to specify the time unit here (SECONDS), because netty documentation did not specify what is the default time unit, and my assumption was ms.
But based on netty code
  https://github.com/netty/netty/blob/21df18deac2d94d8704d6deeb50e9f02f56e6cce/handler/src/main/java/io/netty/handler/timeout/IdleStateHandler.java#L156
        this(readerIdleTimeSeconds, writerIdleTimeSeconds, allIdleTimeSeconds, TimeUnit.SECONDS);

it is seconds. So perhaps we should specify it here, to make it more consistent. It is really easy to mess up time units.

Reply:
Good catch, modified to: 
super(0, connectionKeepAliveTimeOut, 0, TimeUnit.SECONDS);